spinner/examples/loading/main.go

```diff
@@ -8,9 +8,13 @@ import (
 )
 
 func main() {
-	action := func() {
-		time.Sleep(2 * time.Second)
+	action := func() error {
```

Comment:
I usually recommend passing a context on such function


```suggestion
	action := func(ctx context.Context) error {

       select {
         case <- ctx.Done:
             return nil
         case <- time.After(2 * time.Second):
            return nil
       }
```

This way the code can react to a canceled event that occurs before the end if the time.Sleep

Reply:
there are other examples using contexts

Reply:
My point was about passing it to the action

As you seem confident it's a good thing to use a time.Sleep, but maybe it's only for example purpose